website/src/blogs/configuring-openebs-to-run-with-security-enhanced-linux.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Configuring OpenEBS to run with Security-Enhanced Linux
+author: Akhil Mohan
+author_info: Software Engineer @ MayaData, working on Cloud Native Tech.
+excerpt: In this short How-To blog, I will walk you through the steps for running OpenEBS with Security Enhanced Linux (SELinux=on).
+tags: Centos, Kubernetes, OpenEBS, Selinux, Uncategorized, Openshift, Tutorials
+date: 10-05-2019
+---
+
+In this short How-To blog, I will walk you through the steps for running OpenEBS with Security Enhanced Linux (SELinux=on). This applies to both OpenShift based Kubernetes Cluster or Clusters using CentOS.
+
+OpenEBS can be treated as a Cluster Add-on service, that can convert the existing Kubernetes Cluster into a full fledged Storage Controllers — that can serve Persistent Volumes to Stateful Workloads while making use of the storage attached to the nodes. Since OpenEBS runs in containers and has to access the underlying storage devices, it needs to run in Privileged mode.
+
+The component of OpenEBS ( Node Device Manager or Node Disk Manager or simply referred as NDM), is responsible for discovery of the devices and filtering out devices that should not be used by OpenEBS; for example the disk that has OS filesystem. Earlier, to detect the OS disk, the NDM pod by default mounted the `/proc/1/mounts` file, which is restricted on nodes that have SELinux=on. This is now fixed by mounting the `/proc` directory of the host inside the container and then loading the `mounts` file.
+
+So at a high level, to allow OpenEBS to run in privileged mode in SELinux=on nodes, the cluster should be configured to grant privileged access to OpenEBS service account.
+
+Here are the steps I have followed:
+
+****Step 1: Setup appropriate security context for OpenEBS****
+
+**On OpenShift Clusters:** Select the right SCC for OpenEBS
+
+OpenEBS like other cluster add-on services requires its pods to be executed with privileged access as it needs to directly access the block devices. However, this will not mean that SELinux should be turned off.
+
+While running on OpenShift, we need to be aware of the security constraints available and to select the right security constraint for OpenEBS. There are 7 SCCs available.
+
+![Security Context Constraints (SCC) in OpenShift](/images/blog/2019/05/0_nYHmjUbmME9DgkbN.png)
```

Comment:
can we add a relevant name to the image? Also, can we move images directly to `blog` folder, since it's followed for all other blogs

Reply:
Same for other images as well